Include/cpython/interpreteridobject.h

```diff
@@ -8,4 +8,7 @@ PyAPI_DATA(PyTypeObject) PyInterpreterID_Type;
 
 PyAPI_FUNC(PyObject *) PyInterpreterID_New(int64_t);
 PyAPI_FUNC(PyObject *) PyInterpreterState_GetIDObject(PyInterpreterState *);
-PyAPI_FUNC(PyInterpreterState *) PyInterpreterID_LookUp(PyObject *);
```

Comment:
I see that in 3.12 this was a private API (leading `_`) but in 3.13 it lost the `_`. It looks like you're going back to private. What about the handful of definitions above this line? Are any of them meant to be public?

Reply:
That's a good question.  They needn't be public at this point and it turns out interpreteridobject.h was never added to Python.h, so there shouldn't be a problem moving them.  I'll do that here.

Reply:
On second thought, I'm going to remove the `InterpreterID` type, including this header file, in a separate PR.  (I made some changes a few months ago that made the class unnecessary.  It can be removed with a couple small tweaks.)